tests/integration_tests/test_subtensor_integration.py

```diff
@@ -359,6 +361,216 @@ def test_set_weights_failed(self):
         )
         assert fail == False
 
+    def test_commit_weights(self):
+        weights = np.array([0.1, 0.2, 0.3, 0.4], dtype=np.float32)
+        uids = np.array([1, 2, 3, 4], dtype=np.int64)
+        commit_hash = bittensor.utils.weight_utils.generate_weight_hash(
+            who=self.wallet.hotkey.ss58_address,
+            netuid=3,
+            uids=uids.tolist(),
+            values=weights.tolist(),
+            version_key=0,
+        )
+
+        self.subtensor.commit_weights = MagicMock(
```

Comment:
nitpick: It was decided to use pytest without unittest if possible. I suggest using "mocker.MagicMock" instead of "uniitest.mock.MagicMock". I understand that under the hood it's the same thing. But the reference to "mocker" will allow us to get rid of the import of "uniitest.mock".
The below.

Reply:
+1

Reply:
In order to get all of `test_subtensor_integration.py` file away from `unittest.mock.MagicMock` will require a refactor of this whole file. This can and should be done, but not in the scope of this PR.

Reply:
I meant only tests in current PR. Using the mocker fixture does not require additional refactoring or importing.

Reply:
There is no complexity in this - pytest is already used in this module.
You've written new tests using an approach that we want to avoid. This was decided by the team.
Thus, technical debt increases.

But since this was just a nitpick, I shared my opinion with you :)